File: ScoreApi/Models/Judge.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ScoreApi.Models
{
    [Table("Judges")]
    public class Judge
    {
        [Key]
        public int JudgeId { get; set; }

        public int JudgeNumber { get; set; }

        [Required]
        [MaxLength(100)]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [MaxLength(10)]
        public string PinCode { get; set; } = string.Empty;

        [MaxLength(100)]
        public string AssignedRound { get; set; } = string.Empty;

        // Navigation
        public ICollection<ScoreSubmission> ScoreSubmissions { get; set; } = new List<ScoreSubmission>();
    }
}

File: frontend/src/types/judge.ts
export interface Judge {
    judgeId: number;
    judgeNumber: number;
    fullName: string;
    pinCode: string;
    assignedRound: string;
}

export interface CreateJudgeRequest {
    fullName: string;
    assignedRound: string;
}

export interface UpdateJudgeRequest {
    fullName: string;
    assignedRound: string;
}